Makefile

```diff
@@ -185,121 +199,393 @@ test-integration: test-integration-aws test-integration-clickhouse test-integrat
+
+stop-integration-clickhouse:
+	$(CONTAINER_TOOL) rm --force clickhouse 2>/dev/null; true
+ifeq ($(CONTAINER_TOOL),podman)
+	$(CONTAINER_TOOL) $(CONTAINER_ENCLOSURE) stop --name=test-integration-clickhouse 2>/dev/null; true
+	$(CONTAINER_TOOL) $(CONTAINER_ENCLOSURE) rm --force --name test-integration-clickhouse 2>/dev/null; true
+else
+	$(CONTAINER_TOOL) $(CONTAINER_ENCLOSURE) rm test-integration-clickhouse 2>/dev/null; true
 endif
 
 test-integration-clickhouse: ## Runs Clickhouse integration tests
 ifeq ($(AUTOSPAWN), true)
-	${MAYBE_ENVIRONMENT_EXEC} $(CONTAINER_TOOL)-compose up -d dependencies-clickhouse
-	sleep 5 # Many services are very lazy... Give them a sec...
+	$(MAKE) -k stop-integration-clickhouse \
+    ; rc=$$? \
+	$(MAKE) start-integration-clickhouse
+	sleep 5 # Many services are very slow... Give them a sec...
 endif
 	${MAYBE_ENVIRONMENT_EXEC} cargo test --no-default-features --features clickhouse-integration-tests --lib ::clickhouse:: -- --nocapture
 ifeq ($(AUTODESPAWN), true)
-	${MAYBE_ENVIRONMENT_EXEC} $(CONTAINER_TOOL)-compose stop
+	$(MAKE) -k stop-integration-clickhouse
 endif
 
 test-integration-docker: ## Runs Docker integration tests
 	${MAYBE_ENVIRONMENT_EXEC} cargo test --no-default-features --features docker-integration-tests --lib ::docker:: -- --nocapture
 
+start-integration-elasticsearch:
+ifeq ($(CONTAINER_TOOL),podman)
+	$(CONTAINER_TOOL) $(CONTAINER_ENCLOSURE) create --replace --name test-integration-elasticsearch -p 4571:4571 -p 9200:9200 -p 9300:9300 -p 9201:9200 -p 9301:9300
+	$(CONTAINER_TOOL) run -d --$(CONTAINER_ENCLOSURE)=test-integration-elasticsearch --name localstack \
+	 -e SERVICES=elasticsearch:4571 localstack/localstack@sha256:f21f1fc770ee4bfd5012afdc902154c56b7fb18c14cf672de151b65569c8251e
+	$(CONTAINER_TOOL) run -d --$(CONTAINER_ENCLOSURE)=test-integration-elasticsearch \
+	 --name elasticsearch -e discovery.type=single-node -e ES_JAVA_OPTS="-Xms400m -Xmx400m" elasticsearch:6.6.2
+	$(CONTAINER_TOOL) run -d --$(CONTAINER_ENCLOSURE)=test-integration-elasticsearch \
+	 --name elasticsearch-tls -e discovery.type=single-node -e xpack.security.enabled=true \
+	 -e xpack.security.http.ssl.enabled=true -e xpack.security.transport.ssl.enabled=true \
+	 -e xpack.ssl.certificate=certs/localhost.crt -e xpack.ssl.key=certs/localhost.key \
+	 -e ES_JAVA_OPTS="-Xms400m -Xmx400m" \
+	 -v $(PWD)/tests/data:/usr/share/elasticsearch/config/certs:ro elasticsearch:6.6.2
+else
+	$(CONTAINER_TOOL) $(CONTAINER_ENCLOSURE) create test-integration-elasticsearch
+	$(CONTAINER_TOOL) run -d --$(CONTAINER_ENCLOSURE)=test-integration-elasticsearch -p 4571:4571 --name localstack \
+	 -e SERVICES=elasticsearch:4571 localstack/localstack@sha256:f21f1fc770ee4bfd5012afdc902154c56b7fb18c14cf672de151b65569c8251e
+	$(CONTAINER_TOOL) run -d --$(CONTAINER_ENCLOSURE)=test-integration-elasticsearch -p 9200:9200 -p 9300:9300 \
+	 --name elasticsearch -e discovery.type=single-node -e ES_JAVA_OPTS="-Xms400m -Xmx400m" elasticsearch:6.6.2
```

Comment:
These container names should have a `vector_` prefix so contributors can know the source of them.